src/ui/drawing/gdi/ImageRepository.cpp

```diff
@@ -325,14 +325,18 @@ HBITMAP ImageRepository::LoadLocalPNG(const std::wstring& sFilename, size_t nWid
         if (SUCCEEDED(hr))
         {
             if (nWidth == 0 || nHeight == 0)
-                hr = pOriginalBitmapSource->GetSize(&nWidth, &nHeight);
+            {
+                auto nWidth32 = gsl::narrow_cast<std::uint32_t>(nWidth);
+                auto nHeight32 = gsl::narrow_cast<std::uint32_t>(nHeight);
+                hr = pOriginalBitmapSource->GetSize(&nWidth32, &nHeight32);
```

Comment:
`GetSize` parameters are both out. the conversion should be from `nWidth32` to `nWidth`, not the other way around. this code is attempting to determine the width if it's not already known.

Reply:
You don't understand. The point of calling GetSize is to set nWidth and nHeight. You've changed the code to pass in values and ignore the changes, so on line 339 below, nWidth and nHeight will both still be 0!

https://docs.microsoft.com/en-us/windows/desktop/api/wincodec/nf-wincodec-iwicbitmapsource-getsize

Reply:
OK I got you.